data-pipeline/src/agent_info/schema.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+//! This module provides struct representing the info endpoint response
+use serde::Deserialize;
+use std::collections::HashMap;
+
+/// Wrapper for an agent info response storing the state hash from the agent
+#[derive(Clone, Deserialize, Default, Debug, PartialEq)]
+pub struct AgentInfo {
+    /// Hash of the info
+    pub state_hash: String,
+    /// Info response from the agent
+    pub info: AgentInfoStruct,
+}
+
+/// Schema of an agent info response
+#[allow(missing_docs)]
+#[derive(Clone, Deserialize, Default, Debug, PartialEq)]
+pub struct AgentInfoStruct {
+    /// Version of the agent
```

Comment:
Any reason why some of the public fields have doc comments and others don't?

Reply:
The fields are not documented in the agent (or at least I couldn't find it). So I commented the ones that were obvious to me, admittedly not the most useful documentation.